File: applicatie/src/main/java/com/vdzon/administratie/rubriceren/rubriceerRegels/RubriceerPriveRegels.java
package com.vdzon.administratie.rubriceren.rubriceerRegels;

import com.vdzon.administratie.model.Afschrift;
import com.vdzon.administratie.model.Gebruiker;
import com.vdzon.administratie.model.boekingen.PriveBetalingBoeking;
import com.vdzon.administratie.rubriceren.model.RubriceerRegel;

import java.util.UUID;

public class RubriceerPriveRegels extends RubriceerHelper {


    @RubriceerRuleCommit
    public void processRegel(RubriceerRegel regel, Gebruiker gebruiker) {
        Afschrift afschrift = regel.getAfschrift().toAfschrift();
        switch (regel.getRubriceerAction()) {
            case PRIVE:
                PriveBetalingBoeking priveBetalingBoeking = new PriveBetalingBoeking(
                        regel.getAfschrift().getNummer(),
                        UUID.randomUUID().toString());

                gebruiker.getDefaultAdministratie().addBoeking(priveBetalingBoeking);
                break;
        }
    }


}

File: rubriceren/impl/src/main/java/com/vdzon/administratie/rubriceren/rubriceerRegels/RubriceerPriveRegels.kt
package com.vdzon.administratie.rubriceren.rubriceerRegels

import com.vdzon.administratie.model.Afschrift
import com.vdzon.administratie.model.Gebruiker
import com.vdzon.administratie.model.boekingen.PriveBetalingBoeking
import com.vdzon.administratie.rubriceren.model.RubriceerAction
import com.vdzon.administratie.rubriceren.model.RubriceerRegel

import java.util.UUID

class RubriceerPriveRegels : RubriceerHelper() {


    @RubriceerRuleCommit
    fun processRegel(regel: RubriceerRegel, gebruiker: Gebruiker) {
        val afschrift = regel.afschrift.toAfschrift()
        when (regel.rubriceerAction) {
            RubriceerAction.PRIVE -> {
                val priveBetalingBoeking = PriveBetalingBoeking(
                        regel.afschrift.nummer,
                        UUID.randomUUID().toString())

                gebruiker.defaultAdministratie.addBoeking(priveBetalingBoeking)
            }
        }
    }


}
